### PR TITLE
feat: made model equivalence check more rigorous

### PR DESCRIPTION
**Pull request by fsuits**
## Types of changes

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [x] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary
Improved the check for model equivalence to recurse and check every attribute except specified ones.
Converted references to __field_set__ to __field__ because field_set only pertains to fields set when the obect was first initialized.  other fields could be set later and they would be missed.
## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

### CODE REVIEW

File: trestle/common/model_utils.py [hunk 1]
```diff
@@ -837,70 +838,50 @@ def regenerate_uuids(object_of_interest: Any) -> Tuple[Any, Dict[str, str], int]
...
-        return True
+        if obj_a_type is str:
+            if obj_a != obj_b:
+                return True
+        elif isinstance(obj_a, BaseModel):
+            fields_a = getattr(obj_a, '__fields__', None)
+            fields_b = getattr(obj_b, '__fields__', None)
+            if fields_a != fields_b:
+                return True
+            for field in as_list(fields_a):
```
Comment (enikonovad) [inline trestle/common/model_utils.py:863]: Please use as_filtered_list

File: trestle/common/model_utils.py [hunk 2]
```diff
@@ -837,70 +838,50 @@ def regenerate_uuids(object_of_interest: Any) -> Tuple[Any, Dict[str, str], int]
...
-                    return False
-                if len(as_list(cimp_a.implemented_requirements)) != len(as_list(cimp_b.implemented_requirements)):
-                    return False
-                for kk, imp_a in enumerate(as_list(cimp_a.implemented_requirements)):
-                    if not ModelUtils.imp_reqs_are_equivalent(imp_a, cimp_b.implemented_requirements[kk]):
-                        return False
-        return True
+        if obj_a_type is str:
+            if obj_a != obj_b:
+                return True
```
Comment (enikonovad) [inline trestle/common/model_utils.py:857]: return obj_a != obj_b

File: trestle/common/model_utils.py [hunk 3]
```diff
@@ -837,70 +838,50 @@ def regenerate_uuids(object_of_interest: Any) -> Tuple[Any, Dict[str, str], int]
...
-            state_b.uuid = b_uuid
-            if not statements_equiv:
-                return False
-        return True
-
-    @staticmethod
-    def component_defs_are_equivalent(
-        model_a: Optional[comp.ComponentDefinition], model_b: Optional[comp.ComponentDefinition]
-    ) -> bool:
+    def _objects_differ(obj_a: Any, obj_b: Any, ignore_type_list: List[Any], ignore_name_list: List[str]) -> bool:
```
Comment (enikonovad) [inline trestle/common/model_utils.py:841]: Please add unit test

**Review by enikonovad — CHANGES_REQUESTED** [review]
Please add unit tests

File: trestle/common/model_utils.py (hunk 1 above)
Comment (fsuits) [reply trestle/common/model_utils.py:863]: done

File: trestle/common/model_utils.py (hunk 2 above)
Comment (fsuits) [reply trestle/common/model_utils.py:857]: done

File: trestle/common/model_utils.py (hunk 3 above)
Comment (fsuits) [reply trestle/common/model_utils.py:841]: done

**Comment by fsuits** [comment]
I put many of the __fields_set__ back because they are ok - and added new function fields_set_non_none to provide list of field attributes that are not None.

**Review by enikonovad — APPROVED** [review]
LGTM

**Pull request merged**